drivers/lte_link_control/lte_lc.c

```diff
@@ -19,14 +19,37 @@
 
 LOG_MODULE_REGISTER(lte_lc, CONFIG_LTE_LINK_CONTROL_LOG_LEVEL);
 
-#define LC_MAX_READ_LENGTH		128
-#define AT_CMD_SIZE(x)			(sizeof(x) - 1)
-#define AT_CEREG_5			"AT+CEREG=5"
-#define AT_CEREG_READ			"AT+CEREG?"
-#define AT_CEREG_PARAMS_COUNT		10
-#define AT_CEREG_ACTIVE_TIME_INDEX	8
-#define AT_CEREG_TAU_INDEX		9
-#define AT_CEREG_RESPONSE_MAX_LEN	80
+#define LC_MAX_READ_LENGTH			128
+#define AT_CMD_SIZE(x)				(sizeof(x) - 1)
+#define AT_RESPONSE_PREFIX_INDEX		0
+#define AT_CFUN_READ				"AT+CFUN?"
+#define AT_CFUN_RESPONSE_PREFIX			"+CFUN"
+#define AT_CFUN_MODE_INDEX			1
+#define AT_CFUN_PARAMS_COUNT			2
+#define AT_CFUN_RESPONSE_MAX_LEN		20
+#define AT_CEREG_5				"AT+CEREG=5"
+#define AT_CEREG_READ				"AT+CEREG?"
+#define AT_CEREG_RESPONSE_PREFIX		"+CEREG"
+#define AT_CEREG_PARAMS_COUNT			10
+#define AT_CEREG_REG_STATUS_INDEX		1
+#define AT_CEREG_ACTIVE_TIME_INDEX		8
+#define AT_CEREG_TAU_INDEX			9
+#define AT_CEREG_RESPONSE_MAX_LEN		80
+#define AT_XSYSTEMMODE_READ			"AT%XSYSTEMMODE?"
+#define AT_XSYSTEMMODE_RESPONSE_PREFIX		"%XSYSTEMMODE"
+#define AT_XSYSTEMMODE_PROTO			"AT%%XSYSTEMMODE=%d,%d,%d,%d"
```

Comment:
Just a generic note on such proto defines, these should be used with care as %d assumes that the value going in is only one byte ASCII (ie: 0 to 9). If the input (or expected return) is two bytes, its going to be an overflow if you use buf[sizeof(AT_XSYSTEMMODE_PROTO)];

Reply:
I see your point. As you mention, the buffer size is sufficient here, but happy to take suggestions on how to improve this.